### PR TITLE
ROX-20136: Reduce cost of resource alert processing

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

Scope the merging for resource requests

```
     New PR:
     central_active=# select total_exec_time, mean_exec_time, rows, calls, rows/calls as avg, query from pg_stat_statements where query ilike 'select%alerts%' order by calls desc limit 20;
   total_exec_time    |    mean_exec_time    | rows  | calls | avg |                                                                                                                         query                                                                       
                                                   
----------------------+----------------------+-------+-------+-----+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
---------------------------------------------------
   49104.993686999915 |   13.168408068382996 | 29816 |  3729 |   7 | select "alerts".serialized from alerts where (alerts.ClusterId = $1 and (alerts.LifecycleStage = $2) and alerts.Namespace = $3 and alerts.Resource_Name = $4 and (alerts.Resource_ResourceType = $5)
 and ((alerts.State = $6) or (alerts.State = $7)))
(6 rows)

Master:
central_active=# select total_exec_time, mean_exec_time, rows, calls, rows/calls as avg, query from pg_stat_statements where query ilike 'select%alerts%' order by calls desc limit 20;
   total_exec_time   |   mean_exec_time    |   rows   | calls |  avg  |                                                                                                                                                                                                  
                              query                                                                                                                                                                                                                                 
---------------------+---------------------+----------+-------+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
   363841.2265790002 |  1095.9073089728913 | 27506200 |   332 | 82850 | select "alerts".serialized from alerts where (alerts.ClusterId = $1 and (alerts.LifecycleStage = $2) and (alerts.Resource_ResourceType = $3 or alerts.Resource_ResourceType = $4 or alerts.Resour
ce_ResourceType = $5 or alerts.Resource_ResourceType = $6 or alerts.Resource_ResourceType = $7 or alerts.Resource_ResourceType = $8 or alerts.Resource_ResourceType = $9 or alerts.Resource_ResourceType = $10) and ((alerts.State = $11) or (alerts.State = $12)))
(4 rows)
```

The query time and the number of alerts is greatly reduced even without an index

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

See perf test above

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by dashrews78 — APPROVED** [review]
looks good from my perspective regarding the querying.  Should probably wait on an approval from someone more familiar with this alerts in general.

**Review by md2119 — APPROVED** [review]
LGTM

**Review by theencee — APPROVED** [review]

**Pull request merged**
